stac_api/utils/dependencies.py

```diff
@@ -32,27 +25,19 @@ def _parse(request: Request):
     return _parse
 
 
-def database_reader_factory() -> Session:
+def database_reader_factory(request: Request) -> Session:
     """Instantiate the database reader session"""
     try:
-        if not DB_READER:
-            raise DatabaseConnectionError(
-                message="Database engine has not been created"
-            )
-        db = DB_READER()
+        db = request.app.state.DB_READER()
         yield db
     finally:
```

Comment:
Can probably remove the `try...finally` as well and close the db connection during the shutdown event.

Reply:
Which should also let us remove the reader/writer dependencies all together by updating the backend clients to pull the database session directly from the request:
- https://github.com/vincentsarago/arturo-stac-api/blob/storeDepsInState/stac_api/clients/item_crud.py#L134-L135
- https://github.com/vincentsarago/arturo-stac-api/blob/storeDepsInState/stac_api/clients/collection_crud.py#L66-L67
- https://github.com/vincentsarago/arturo-stac-api/blob/storeDepsInState/stac_api/clients/tokens.py#L40-L41